Orange/canvas/application/canvastooldock.py

```diff
@@ -87,33 +89,39 @@ def setSplitterAndWidget(self, splitter, widget):
         if splitter and widget and not splitter.indexOf(widget) > 0:
             raise ValueError("Widget must be in a spliter.")
 
-        if self.__widget:
-            self.__widget.removeEventFilter()
+        if self.__widget is not None:
+            self.__widget.removeEventFilter(self)
+        if self.__splitter is not None:
+            self.__splitter.removeEventFilter(self)
+
         self.__splitter = splitter
         self.__widget = widget
 
-        if widget:
+        if widget is not None:
```

Comment:
I am not sure how the document relates to the preference. The document explicitly states:

`Beware of writing if x: when you really mean if x is not None`